src/librbd/ManagedLock.cc

```diff
@@ -60,18 +60,24 @@ ManagedLock<I>::~ManagedLock() {
 }
 
 template <typename I>
-bool ManagedLock<I>::is_lock_owner() const {
+bool ManagedLock<I>::is_lock_owner(bool exclusive) const {
   Mutex::Locker locker(m_lock);
 
   bool lock_owner;
 
   switch (m_state) {
-  case STATE_LOCKED:
+  case STATE_SHARED_LOCKED:
```

Comment:
Since you already have ```m_last_lock_type_exclusive``` (nit: perhaps rename that to something like ```m_exclusively_locked```) and are using it all over the place to determine if you should be doing an exclusive vs shared lock, why not re-use it here and collapse all the new states back down into there original states. You can keep the SHARED vs EXCLUSIVE action states.